[PyTorch] SDPA decomp: actually use attn_mask

digantdesai opened the pull request:
Summary: Need to pass this along

Test Plan:
```
cd ~/fbsource/fbcode/executorch/backends/xnnpack/test
buck test fbcode//mode/dev-nosan :test_xnnpack_ops -- test_fp32_sdpa
buck run fbcode//mode/dev-nosan :test_xnnpack_models -- executorch.backends.xnnpack.test.models.llama2_et_example.TestLlama2ETExample.test_fp32
```

Reviewed By: larryliu0820

Differential Revision: D52812369

larryliu0820 commented on torch/_decomp/decompositions.py:
```diff
@@ -4321,7 +4321,14 @@ def scaled_dot_product_flash_attention_for_cpu(
+        query,
+        key,
+        value,
+        attn_mask=attn_mask,
```
Can you update the test in https://github.com/pytorch/pytorch/blob/main/test/test_decomp.py#L921

larryliu0820 reviewed (approved):
Can you add a unit test

digantdesai committed:
[PyTorch] SDPA decomp: actually use attn_mask (#117579)

Summary:

Need to pass this along

Test Plan:
```
cd ~/fbsource/fbcode/executorch/backends/xnnpack/test
buck test fbcode//mode/dev-nosan :test_xnnpack_ops -- test_fp32_sdpa_mask2d
buck run fbcode//mode/dev-nosan :test_xnnpack_models -- executorch.backends.xnnpack.test.models.llama2_et_example.TestLlama2ETExample.test_fp32
```

Reviewed By: larryliu0820

Differential Revision: D52812369
